src/main/java/com/redhat/devtools/lsp4ij/internal/CompletableFutures.java

```diff
@@ -148,27 +138,27 @@ public static void waitUntilDone(@NotNull CompletableFuture<?> future,
      * Wait in Task (which is cancellable) for the done of the given future and stop the wait if {@link ProcessCanceledException} is thrown.
      *
      * @param future the future to wait.
-     * @param title the task title.
-     * @param file the Psi file.
+     * @param title  the task title.
+     * @param file   the Psi file.
      */
     public static void waitUntilDoneAsync(@NotNull CompletableFuture<?> future,
-                                     @NotNull String title,
-                                     @NotNull PsiFile file) {
+                                          @NotNull String title,
+                                          @NotNull PsiFile file) {
 
         ProgressManager.getInstance().run(new Task.Backgroundable(file.getProject(), title, true) {
             @Override
             public void run(@NotNull ProgressIndicator indicator) {
                 try {
                     waitUntilDone(future, file);
                 } catch (CancellationException | ProcessCanceledException e) {
-                    // Do nothing
-                }
-                catch(Exception e) {
-                    // Should never occur
-                    LOGGER.error("Error while executing future '" + title + "'.", e);
-                }
-                finally {
-                    indicator.stop();

```

Comment:
why not keep that finally block?

Reply:
We dont need it according my test with prepare rename